src/esp/sensor/SensorFactory.h

```diff
@@ -6,16 +10,33 @@ namespace sensor {
 class SensorFactory {
  public:
   /**
-   * @brief Static method to initialize and return a SensorSuite of Sensors
-   * @param[in] node SceneNode to create child SceneNodes and attach sensors to,
-   * sensorSetup SensorSetup a vector of SensorSpec::ptr defining specs for each
-   * sensor
-   * @return SensorSuite of specified Sensors attached to children SceneNodes of
-   * node
+   * @brief Static method to initialize SensorSuite of Sensors
+   * @param[in] node SceneNode of SensorSuite to create child SceneNodes and
+   * attach sensors to
+   * @param[in] sensorSetup SensorSetup a vector of SensorSpec::ptr defining
+   * specs for each sensor
+   *
+   */
+  static std::map<std::string, std::reference_wrapper<sensor::Sensor>>&
+  createSensors(scene::SceneNode& node, const sensor::SensorSetup& sensorSetup);
+
+  /**
+   * @brief Static method to delete a Sensor
+   * @param[in] sensor Sensor to delete
+   *
+   */
+  static void deleteSensor(const sensor::Sensor& sensor);
+
+  /**
+   * @brief Static method to delete a Sensor from the subtree rooted at node
+   * @param[in] node root SceneNode of the subtree in which the Sensor will be
+   * searched
+   * @param[in] uuid Sensor's uuid
+   * NOTE: If sensor does not exist, SensorFactory will log a message but
+   * proceed without any errors thrown
+   *
    */
-  static sensor::SensorSuite createSensors(
-      scene::SceneNode& node,
-      const sensor::SensorSetup& sensorSetup);
+  static void deleteSensor(scene::SceneNode& node, const std::string& uuid);
```

Comment:
Just read the comments. So consider call it `deleteSubtreeSensor` to avoid confusion.